packages/legend-application-studio/src/components/editor/editor-group/connection-editor/RelationalDatabaseConnectionEditor.tsx

```diff
@@ -1589,11 +1589,21 @@ const RelationalConnectionGeneralEditor = observer(
       val: { label: string; value: string } | null,
     ): void => {
       dBConnection_setType(connection, val?.value ?? DatabaseType.H2);
+      if (connectionValueState.selectedValidDatasources[0]) {
+        connectionValueState.changeDatasourceSpec(
+          connectionValueState.selectedValidDatasources[0],
+        );
+      }
+      if (connectionValueState.selectedValidAuthenticationStrategies[0]) {
+        connectionValueState.changeAuthenticationStrategy(
+          connectionValueState.selectedValidAuthenticationStrategies[0],
+        );
+      }
     };
 
     // source spec type
     const sourceSpecOptions = (
-      Object.values(CORE_DATASOURCE_SPEC_TYPE) as string[]
```

Comment:
as discussed offline, remove unused imports of `CORE_DATASOURCE_SPEC_TYPE` and `CORE_AUTHENTICATION_STRATEGY_TYPE`